packages/model-viewer/src/three-components/Reticle.ts

```diff
@@ -47,42 +49,45 @@ export default class Reticle extends Object3D {
     this.camera = camera;
   }
 
+  get hitMatrix(): Matrix4|null {
+    return this._hitMatrix;
+  }
+
   /**
    * Fires a hit test in the middle of the screen and places the reticle
    * upon the surface if found.
    *
-   * @param {XRSession} session
+   * @param {XRFRame} frame
    * @param {XRFrameOfReference} frameOfRef
    */
-  async update(session: XRSession, frameOfRef: XRFrameOfReference) {
-    this.raycaster = this.raycaster || new Raycaster();
-    this.raycaster.setFromCamera({x: 0, y: 0}, this.camera);
-    const ray = this.raycaster.ray;
-    let xrray: XRRay = new XRRay(ray.origin, ray.direction);
-
-    let hits: Array<XRHitResult>;
-
-    try {
-      hits = await session.requestHitTest(xrray, frameOfRef);
-    } catch (error) {
-      hits = [];
-    }
-
-    if (hits.length) {
-      const hit = hits[0];
-      const hitMatrix = new Matrix4().fromArray(hit.hitMatrix as any);
+  async update(_session: XRSession, _frame: XRFrame, _viewerReferenceSpace: XRFrameOfReference, _frameOfRef: XRFrameOfReference) {
+    if(!this.hitTestSourceRequest) {
+      this.hitTestSourceRequest = _session.requestHitTestSource({space: _viewerReferenceSpace as any})
+                                          .then(hitTestSource => {
+                                            this.hitTestSource = hitTestSource;
+                                          });
+    } else if(this.hitTestSource){
+      const hitTestResults = _frame.getHitTestResults(this.hitTestSource);
+      if (hitTestResults.length) {
+        const hit = hitTestResults[0];
+        this._hitMatrix = new Matrix4().fromArray(hit.getPose(_frameOfRef as any)!.transform.matrix);
```

Comment:
I'm noticing a number of `as any`; can we get proper TS types for these somewhere? I know they weren't there before either, but we are trying to come out of unstable.

Reply:
We already have typings for what those objects are (`XRReferenceSpace`s) in webxr.ts - the problem here is that ARRenderer.ts has both the $viewerRefSpace and $refSpace typed as `XRCoordinateSystem`. If you think that those can be safely changed (I have not tried doing so) from XRCoordinateSystem into XRReferenceSpace, then there should be no need for so many casts to `any` (both here and in ARRenderer.tx).

Reply:
Let's give it a shot. If it doesn't work, that'll be good info to have. This stuff is all pretty separate from the rest of the code base, so I'm not worried about far-reaching implications.

Reply:
Removed all WebXR-related `any` casts from Reticle.ts and ARRenderer.ts.